xml/array_test.go

```diff
@@ -0,0 +1,56 @@
+package xml
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestWrappedArray(t *testing.T) {
+	buffer := bytes.NewBuffer(nil)
+	scratch := make([]byte, 64)
+	a := newArray(buffer, &scratch, nil, arrayMemberWrapper)
+	a.Member().String("bar")
+	a.Member().String("baz")
+
+	e := []byte(`<member>bar</member><member>baz</member>`)
+	if a := buffer.Bytes(); bytes.Compare(e, a) != 0 {
+		t.Errorf("expected %+q, but got %+q", e, a)
+	}
+}
+
+func TestWrappedArrayWithCustomName(t *testing.T) {
+	buffer := bytes.NewBuffer(nil)
+	scratch := make([]byte, 64)
+
+	a := newArray(buffer, &scratch, nil, "item")
+	a.Member().String("bar")
+	a.Member().String("baz")
+
+	e := []byte(`<item>bar</item><item>baz</item>`)
+	if a := buffer.Bytes(); bytes.Compare(e, a) != 0 {
+		t.Errorf("expected %+q, but got %+q", e, a)
+	}
+}
+
+func TestFlattenedArray(t *testing.T) {
+	buffer := bytes.NewBuffer(nil)
+	scratch := make([]byte, 64)
+
+	startElement := StartElement{
+		Name: Name{
+			Local: "flattened",
+		},
+	}
+
+	endElement := startElement.End()
+
+	a := newFlattenedArray(buffer, &scratch, &startElement, &endElement)
+
+	a.Member().String("bar")
+	a.Member().String("bix")
+
+	e := []byte(`<flattened>bar</flattened><flattened>bix</flattened>`)
+	if a := buffer.Bytes(); bytes.Compare(e, a) != 0 {
+		t.Errorf("expected %+q, but got %+q", e, a)
+	}
+}
```

Comment:
Whats the difference between a flattened array and non-flattened array in these tests. Other than having a custom name it doesn't look like there is any difference between the two?  Can these two be collapsed?